Lib/typing.py

```diff
@@ -1136,8 +1136,18 @@ def __mro_entries__(self, bases):
         if self.__origin__ not in bases:
             res.append(self.__origin__)
         i = bases.index(self)
+        # The goal here is to only add Generic to the MRO if nothing else in the
+        # MRO is already a subclass of Generic; otherwise we risk failure to
+        # linearize a consistent MRO.
         for b in bases[i+1:]:
-            if isinstance(b, _BaseGenericAlias) or issubclass(b, Generic):
+            if isinstance(b, _BaseGenericAlias):
+                break
+            if not isinstance(b, type):
+                meth = getattr(b, "__mro_entries__", None)
```

Comment:
Nut: I'd probably call this variable mro_entries, `mro_entries_meth`, or `entries_meth`

Reply:
But too long name will double increase the length of the next line.

Reply:
> But too long name will double increase the length of the next line.

If the next line is changed to

```py
                new_bases = entries_meth(bases) if entries_meth else None
```

It will still only be 74 characters long, well within the 80-character limit PEP-8 recommends

Reply:
But if you use `mro_entries_meth` and add `is not None` in the test condition...

Reply:
I'm not sure what point you're trying to make :) I agree we generally shouldn't go over 80 lines, but I don't see that as a good reason to insist on ambiguous variable names.

Of the three I suggested, I'd probably go for `entries_meth`. If you changed the condition to `is not None`, you could easily still keep it within 80 lines like this

```py
                new_bases = (
                    entries_meth(bases)
                    if entries_meth is not None
                    else None
                )
```

But anyway, this one is just personal preference, really. Just a nitpick :)

Reply:
My point is that `meth` is not so bad name. There are many examples of using it in the Python and C code.

Although I prefer more concrete names too, this case is not important enough. Any name is good if not too extreme.

Reply:
I don't disagree with any of your points; I think we're just weighing the balance slightly differently :)

As I said, this was just a nitpick; I'm not insisting. I agree that `meth` isn't _that_ bad a name.